### PR TITLE
:rocket: Naive Hybrid Update Scheme for mVLSI

### PR DESCRIPTION
**Pull request by micheltakken**

### CODE REVIEW

File: src/hybridDynamics/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+set(SOURCE_LIST
+    Scheme.hh
```
Comment (micheltakken) [inline src/hybridDynamics/CMakeLists.txt:2]: Missing Naive.hh

File: src/hybridDynamics/Naive.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/**
+ * @file Scheme.h
```
Comment (micheltakken) [inline src/hybridDynamics/Naive.h:2]: Naive.h

File: src/hybridDynamics/Naive.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+namespace mmft{
+
+/**
+ * @brief Update scheme that defines the update rules between Abstract and CFD for Hybrid simulation.
+ */
+template<typename T>
+class NaiveScheme : public Scheme<T> {
+
+public:
+
```
Comment (micheltakken) [inline src/hybridDynamics/Naive.h:27]: Add comments

File: src/hybridDynamics/Scheme.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+class Scheme {
+
+protected:
+
+    std::unordered_map<int, T> alpha;       // update pressure values on the openings, i.e., Abstract-CFD interface nodes <nodeId, alpha>
+    std::unordered_map<int, T> beta;        // update flow rate values on the openings, i.e., Abstract-CFD interface nodes <nodeId, alpha>
+    std::unordered_map<int, int> theta;     // Amount of LBM collide and stream iterations between update steps <moduleId, theta>
+
+    Scheme();
+
```
Comment (micheltakken) [inline src/hybridDynamics/Scheme.h:33]: Add comments

File: src/nodalAnalysis/CMakeLists.txt
Comment (micheltakken) [inline src/nodalAnalysis/CMakeLists.txt:1]: Reverse changes

File: src/nodalAnalysis/NodalAnalysis.h
Comment (micheltakken) [inline src/nodalAnalysis/NodalAnalysis.h:1]: Revert changes

File: src/nodalAnalysis/NodalAnalysis.hh
Comment (micheltakken) [inline src/nodalAnalysis/NodalAnalysis.hh:1]: Revert changes

File: src/porting/jsonReaders.hh [hunk 5]
```diff
@@ -249,12 +249,13 @@ void readSimulators(json jsonString, sim::Simulation<T>& simulation, arch::Netwo
         } else {
             vtkFolder = "./tmp/";
         }
+        T alpha = jsonString["simulation"]["settings"]["simulators"][0]["alpha"];
+        simulation.setHybridScheme(alpha, 5*alpha, 10);
```
Comment (micheltakken) [inline src/porting/jsonReaders.hh:253]: Remove hardcoded values

Comment (micheltakken) [inline src/porting/jsonReaders.hh:253]: Also, the hybrid scheme must be defined from the JSON file

File: src/simulation/CFDSim.h [hunk 6]
```diff
@@ -18,7 +18,7 @@ class CFDSimulator;
      * @param[in] network The network on which the CFD simulations are conducted.
      */
     template<typename T>
-    bool conductCFDSimulation(const std::unordered_map<int, std::unique_ptr<CFDSimulator<T>>>& cfdSimulators, int iteration);
+    bool conductCFDSimulation(const std::unordered_map<int, std::unique_ptr<CFDSimulator<T>>>& cfdSimulators, int iteration, T alpha);
 
     template<typename T>
```
Comment (micheltakken) [inline src/simulation/CFDSim.h:23]: Add comments

File: src/simulation/Simulation.h [hunk 7]
```diff
@@ -13,6 +13,11 @@
 #include <unordered_map>
 #include <vector>
 
+#include "Eigen/Dense"
```
Comment (micheltakken) [inline src/simulation/Simulation.h:16]: Remove unnecessary heading

File: src/simulation/Simulation.h [hunk 8]
```diff
@@ -192,6 +208,8 @@ class Simulation {
     */
     void saveMixtures();
 
+    void storeErrors();
```
Comment (micheltakken) [inline src/simulation/Simulation.h:211]: Move to better location

File: src/simulation/Simulation.h [hunk 9]
```diff
@@ -277,6 +295,8 @@ class Simulation {
      */
     Mixture<T>* addDiffusiveMixture(std::unordered_map<int, Specie<T>*> species, std::unordered_map<int, std::tuple<std::function<T(T)>, std::vector<T>, T>> specieDistributions);
```
Comment (micheltakken) [inline src/simulation/Simulation.h:297]: Add Comments

File: src/simulation/Simulation.hh [hunk 10]
```diff
@@ -524,6 +530,29 @@ namespace sim {
         return newDroplet;
     }
 
+    template<typename T>
```
Comment (micheltakken) [inline src/simulation/Simulation.hh:533]: Move to better location

File: src/simulation/simulators/cfdSimulator.h [hunk 11]
```diff
@@ -52,7 +59,7 @@ class CFDSimulator {
 
 public:
```
Comment (micheltakken) [inline src/simulation/simulators/cfdSimulator.h:61]: Add comment

File: src/simulation/simulators/olbContinuous.h [hunk 12]
```diff
@@ -49,7 +49,7 @@ using BounceBack = olb::BounceBack<T,DESCRIPTOR>;
     int step = 0;                           ///< Iteration step of this module.
     int stepIter = 1000;                    ///< Number of iterations for the value tracer.
     int maxIter = 1e7;                      ///< Maximum total iterations.
-    int theta = 10;                         ///< Number of OLB iterations per communication iteration.
+    int theta = 1;                          ///< Number of OLB iterations per communication iteration.
```
Comment (micheltakken) [inline src/simulation/simulators/olbContinuous.h:52]: Remove this value

**Review by MariaEmmerich — APPROVED** [review]

**Pull request merged**